R-package/R/callback.R

```diff
@@ -37,7 +37,12 @@ cb.reset.parameters <- function(new_params) {
 
     # Some parameters are not allowed to be changed,
     # since changing them would simply wreck some chaos
-    not_allowed <- c("num_class", "metric", "boosting_type")
+    not_allowed <- c(
+      "num_class"
+      , "metric"
+      , "boosting"
+      , .PARAMETER_ALIASES()[["boosting"]]
```

Comment:
I think `boosting` itself can be in returned values, so that only `.PARAMETER_ALIASES()[["boosting"]]` will be needed instead of current `"boosting", .PARAMETER_ALIASES()[["boosting"]]`.

Reply:
you are right. If aliases are all in the cpp library there shouldn't a reason to care (in language wrappers) about what the "main" one is and what the "alias" ones are. I'll fix that.

Reply:
just changed this so now all valid names ("main" or otherwise) are returned